File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int i=0,j=0,n=nums.size();
        int ans=INT_MIN,cur=0;
        while(j<n)
        {
            cur+=nums[j];
            ans=max(ans,cur);
            while(cur<0)
            {
                cur-=nums[i];
                i++;
            }
            j++;
        }
        return ans;
    }
};

File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        # when is -ve then i start new as it is always good to have accumulated somthing 
        max_sum = nums[0]
        curr_sum = 0
        for num in nums:
            if (curr_sum < 0):
                curr_sum = 0
            curr_sum += num
            max_sum = max(max_sum, curr_sum)
        return max_sum 
            
            
        